File: src/main/java/com/forcetracker222/service/SurveillanceDataService.java
package com.forcetracker222.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.forcetracker222.domain.SurveillanceData;
import com.forcetracker222.dto.SurveillanceDataDTO;
import com.forcetracker222.dto.SurveillanceDataSearchDTO;
import com.forcetracker222.dto.SurveillanceDataPageDTO;
import com.forcetracker222.dto.SurveillanceDataConvertCriteriaDTO;
import com.forcetracker222.service.GenericService;
import com.forcetracker222.dto.common.RequestDTO;
import com.forcetracker222.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface SurveillanceDataService extends GenericService<SurveillanceData, Integer> {

	List<SurveillanceData> findAll();

	ResultDTO addSurveillanceData(SurveillanceDataDTO surveillanceDataDTO, RequestDTO requestDTO);

	ResultDTO updateSurveillanceData(SurveillanceDataDTO surveillanceDataDTO, RequestDTO requestDTO);

    Page<SurveillanceData> getAllSurveillanceDatas(Pageable pageable);

    Page<SurveillanceData> getAllSurveillanceDatas(Specification<SurveillanceData> spec, Pageable pageable);

	ResponseEntity<SurveillanceDataPageDTO> getSurveillanceDatas(SurveillanceDataSearchDTO surveillanceDataSearchDTO);
	
	List<SurveillanceDataDTO> convertSurveillanceDatasToSurveillanceDataDTOs(List<SurveillanceData> surveillanceDatas, SurveillanceDataConvertCriteriaDTO convertCriteria);

	SurveillanceDataDTO getSurveillanceDataDTOById(Integer surveillanceDataId);







}

File: forcetracker222_ui/src/services/SurveillanceDataService.js
import http from "../http-common";

class SurveillanceDataService {
  getAllSurveillanceDatas(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/surveillanceData/surveillanceDatas`, searchDTO);
  }

  get(surveillanceDataId) {
    return this.getRequest(`/surveillanceData/${surveillanceDataId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/surveillanceData?field=${matchData}`, null);
  }

  addSurveillanceData(data) {
    return http.post("/surveillanceData/addSurveillanceData", data);
  }

  update(data) {
  	return http.post("/surveillanceData/updateSurveillanceData", data);
  }
  
  uploadImage(data,surveillanceDataId) {
  	return http.postForm("/surveillanceData/uploadImage/"+surveillanceDataId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new SurveillanceDataService();
